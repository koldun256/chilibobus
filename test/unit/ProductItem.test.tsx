import React, { act } from "react";
import { ProductItem } from "../../src/client/components/ProductItem";
import { render, screen } from "@testing-library/react";
import { addToCart, initStore } from "../../src/client/store";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { CartApi, ExampleApi } from "../../src/client/api";

const product = {
  id: 69,
  name: "bipki",
  price: 420,
  description: "lorem",
  material: "ipsum",
  color: "doret",
};
function renderProductItem() {
  localStorage.clear();
  const api = new ExampleApi("/hw/store");
  const cart = new CartApi();
  const store = initStore(api, cart);
  const { container } = render(
    <BrowserRouter>
      <Provider store={store}>
        <ProductItem product={product} />
      </Provider>
    </BrowserRouter>
  );
  return { store, container };
}
describe("Компонент ProductItem", () => {
  it("соответствует снапшоту без корзины", () => {
    const { container } = renderProductItem();

    expect(container).toMatchSnapshot("ProductItem без корзины");
  });
  it("соответствует снапшоту с корзиной", () => {
    const { container, store } = renderProductItem();

    act(() => store.dispatch(addToCart(product)));

    expect(container).toMatchSnapshot("ProductItem с корзиной");
  });
});
