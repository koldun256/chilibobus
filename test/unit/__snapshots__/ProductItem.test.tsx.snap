// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент ProductItem соответствует снапшоту без корзины: ProductItem без корзины 1`] = `
<div>
  <div
    class="ProductItem card w-100 mb-4"
    data-testid="69"
  >
    <img
      class="Image card-img-top"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
    />
    <div
      class="card-body"
    >
      <h5
        class="ProductItem-Name card-title"
      >
        bipki
      </h5>
      <p
        class="ProductItem-Price card-text"
      >
        $
        420
      </p>
      <a
        class="ProductItem-DetailsLink card-link"
        href="/catalog/69"
      >
        Details
      </a>
    </div>
  </div>
</div>
`;

exports[`Компонент ProductItem соответствует снапшоту с корзиной: ProductItem с корзиной 1`] = `
<div>
  <div
    class="ProductItem card w-100 mb-4"
    data-testid="69"
  >
    <img
      class="Image card-img-top"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
    />
    <div
      class="card-body"
    >
      <h5
        class="ProductItem-Name card-title"
      >
        bipki
      </h5>
      <p
        class="ProductItem-Price card-text"
      >
        $
        420
      </p>
      <a
        class="ProductItem-DetailsLink card-link"
        href="/catalog/69"
      >
        Details
      </a>
      <span
        class="CartBadge text-success mx-3"
      >
        Item in cart
      </span>
    </div>
  </div>
</div>
`;
